src/Combat.cpp

```diff
@@ -0,0 +1,27 @@
+#include "Combat.h"
+
+void CombatSystem::on_tick()
+{
+    for(auto&& [attack_entity, attack, attack_pos, sprite, flip] : QueryAttacks :: access_storage().each()){
+        if(flip == None){
+            attack_pos.xy.x += attack.speed * Time::delta_time();
+            if(attack_pos.xy.x > attack.range){
+                despawn(attack_entity);
+            }
+        }
+        else{
+            attack_pos.xy.x -= attack.speed * Time:: delta_time();
+            if(attack_pos.xy.x < attack.range){
+                despawn(attack_entity);
+            }
+        }
+
+        for(auto&& [player_entity, player, player_pos] : QueryPlayers :: access_storage().each()){
+            if (	abs(player_pos.xy.x - attack_pos.xy.x) <= attack.radius
```

Comment:
if you want to align these, i'd rather go for aligning with the start of the line but eh, it might be okay

Reply:
whatever you choose, choose consistently!